unix.cpp

```diff
@@ -2,15 +2,59 @@
 #include "settings.h"
 #include <QDir>
 #include <QStandardPaths>
+#include <QString>
 
 namespace Sys {
 QString archiveName(void)
 {
     return "linux64.zip";
 }
 
+void migrateHomePath(void)
+{
+    QString legacyHomePath = QDir::homePath() + "/.unvanquished";
+    QString xdgDataHome = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
+    QString xdgHomePath = xdgDataHome + "/unvanquished";
+
+    if (QDir(legacyHomePath).exists()) {
+        if (!QDir(xdgHomePath).exists()) {
+            if (!QDir(xdgDataHome).exists()) {
+                if (!QDir().mkdir(xdgDataHome)) {
+                    qFatal("Could not create XDG data directory %s", qPrintable(xdgDataHome));
+                }
+            }
+
+            QFileInfo fileinfo(legacyHomePath);
+            if (fileinfo.isSymLink()) {
```

Comment:
Renaming symlinks doesn't work? Are you going to delete the old symlink?

Reply:
Moving symlink is a non-solvable problem. Moving a relative symlink is very likely to break it. You can also try to read the symlink destination if relative and try to reproduce the relativeness but it ends to yet another undecidable problem since you can't know what the user want (and there can be multiple symlink in the path) and the required level of relativeness. The only safe option is to symlink the original symlink, that's the only option that can't break something. Note that if the homepath is already a symlink the user is an advanced one and already knows what he did and knows what to do if he wants a better solution than a symlink to a symlink.